File: DataStructures/BreadthFirst/Node.cpp
#include "Node.h"
Node::Node(char val) {
	this->value = val;
}
Node::~Node() {
	for (int i = 0; i < this->neighbors.size();i++) {
		if (this->neighbors[i] != nullptr) {
			delete this->neighbors[i];
		}
		this->neighbors[i]->to = nullptr;
		this->neighbors[i] = nullptr;
	}
}
void Node::AddRoute(Node* node) {
	Edge* neighbor = new Edge(node);
	this->neighbors.push_back(neighbor);
}

File: src/graph/breadthFirst/node.ts
import {INode} from "../INode";
import Edge from "./edge";
export default class Node implements INode<string>{
    value: string;
    neighbors: Array<Edge>;
    constructor(val:string) {
        this.value = val;
        this.neighbors = [];
    }

    addRoute(node: INode<string>): void {
        let neighbor = new Edge(node);
        this.neighbors.push(neighbor);
    }
}